python_client/ambu_control.py

```diff
@@ -160,14 +161,19 @@ def _handleSerial(self):
                     if doNotify and self._confCallBack is not None:
                         self._confCallBack()
 
-                elif data[0] == 'STATUS' and len(data) == 5:
+                elif data[0] == 'STATUS' and len(data) == 6:
                     #print(f"Got status: {line.rstrip()}")
-                    count = int(data[1],0)
-                    press = float(data[2])
-                    flow  = float(data[3])
-                    vol   = float(data[4])
-                    diffT = ts - self._stime
-
+                    millis = int(data[1],0) 
+                    count  = int(data[2],0)
+                    press  = float(data[3])
+                    flow   = float(data[4])
+                    vol    = float(data[5])
+                    #diffT  = ts - self._stime
+                    if self._smillis==-1: 
+                        self._smillis=millis
+                        continue
+                    else:
+                        diffT=(millis-self._smillis)/1000.
```

Comment:
I don't this this is enough. this may result in a negative number.

Reply:
Noted: it should have some roll-over handling here. In principle just discarding a negative interval every 50 days is probably fine.